File: LLD/InterviewExamples/Blinkit/Java/src/repositories/ProductRepository.java
package repositories;

import models.Product;

import java.util.List;

public class ProductRepository {

    private List<Product> productList;

    public ProductRepository(List<Product> productList) {
        this.productList = productList;
    }

    public List<Product> getProductList() {
        return this.productList;
    }

    public Product getProductByName(String name) {
        for(Product product : productList) {
            if(product.getName().equals(name)) {
                return product;
            }
        }
        return null;
    }
}

File: LLD/InterviewExamples/Blinkit/Node/src/repositories/productRepository.js
const Product = require("../models/product");

const productList = [
    new Product("Shirt","CLOTHING",100),
    new Product("Vicks","PHARMECEUTICAL",30)
]


module.exports = {
    addProduct : (product) => productList.push(product),
    getProductList : () => productList
}